test/functional/mempool_accept.py

```diff
@@ -58,38 +55,30 @@ def check_mempool_result(self, result_expected, *args, **kwargs):
 
     def run_test(self):
         node = self.nodes[0]
+        miniwallet = MiniWallet(node)
 
-        self.log.info('Start with empty mempool, and 200 blocks')
+        self.log.info('Start with empty mempool, and 400 blocks')
         self.mempool_size = 0
-        assert_equal(node.getblockcount(), 200)
+        miniwallet.generate(200)
+        assert_equal(node.getblockcount(), 400)
         assert_equal(node.getmempoolinfo()['size'], self.mempool_size)
-        coins = node.listunspent()
 
         self.log.info('Should not accept garbage to testmempoolaccept')
         assert_raises_rpc_error(-3, 'Expected type array, got string', lambda: node.testmempoolaccept(rawtxs='ff00baar'))
         assert_raises_rpc_error(-8, 'Array must contain exactly one raw transaction for now', lambda: node.testmempoolaccept(rawtxs=['ff00baar', 'ff22']))
         assert_raises_rpc_error(-22, 'TX decode failed', lambda: node.testmempoolaccept(rawtxs=['ff00baar']))
 
         self.log.info('A transaction already in the blockchain')
-        coin = coins.pop()  # Pick a random coin(base) to spend
-        raw_tx_in_block = node.signrawtransactionwithwallet(node.createrawtransaction(
-            inputs=[{'txid': coin['txid'], 'vout': coin['vout']}],
-            outputs=[{node.getnewaddress(): 0.3}, {node.getnewaddress(): 49}],
-        ))['hex']
-        txid_in_block = node.sendrawtransaction(hexstring=raw_tx_in_block, maxfeerate=0)
-        node.generate(1)
+        tx_in_block = miniwallet.send_self_transfer(from_node = node, default_fee = True)
+        miniwallet.generate(1) # Generate a block, tx gets mined therefore mempool is empty
```

Comment:
Any reason to change this from `node.generate(1)`? `miniwallet.generate` will put an immature utxo in the wallet.